02_commands.html

```diff
@@ -1,206 +1,524 @@
+
+    <ol>
+      <li><p>Let <var>response</var> be a new <a href="https://fetch.spec.whatwg.org/#concept-response" class="externalRef">response</a>.</p></li>
+
+      <li><p>Set <var>response</var>'s <a href="https://fetch.spec.whatwg.org/#concept-response-status" class="externalRef">status</a> to <var>status</var>, and <a href="https://fetch.spec.whatwg.org/#concept-response-status-message" class="externalRef">status message</a> to the string corresponding to the description of <var>status</var> in the <a href="http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml" class="externalRef">status code registry</a>.</p></li>
+      <li><p>If <var>data</var> is not null, let <var>response</var>'s <a href="https://fetch.spec.whatwg.org/#concept-response-body" class="externalRef">body</a> be the result of <a>serializing as JSON</a> with <var>data</var> as the argument.</p></li>
+
+      <li><p>Serialize <var>response</var> according to the rules in [[!RFC7230]] and write the resulting bytes to the <a>connection</a></p></li>
+
+    </ol>
+
+    <p>When required to <dfn>return an error</dfn>, with <var>error code</var>, run the following steps:</p>
+
+    <ol>
+      <li><p>Let <var>http status</var> and <var>name</a> be the <a>error response data</a> for <var>error code</var></p></li>
+      <li><p>Let <var>data</var> be a <a>new JSON Object</a>.</p>
+
+      <li><p>Let <var>message</var> be an implementation-defined string containing a human-readable description of the reason for the error.</p></li>
+
+        <!-- TODO: really need a better way of constructing JSON literals -->
+      <li><p><a>Set the properties</a> of <var>object</var> with values ("error", <var>name</var>), ("message", <var>message</var>).</p></li>
+
+      <li><a>Send a response</a> with status <var>status</var> and data <var>data</var></li>
+
+    </ol>
+
+  </section>
+
+  <section>
+    <h3>Routing Requests</h3>
+
+    <!-- TODO: define what a command is -->
+
+    <p>Request routing is the process of going from a HTTP request to the series of steps needed to implement the <a>command</a> represented by that request.</p>
+
+    <p>A <a>remote end</a> may have a <dfn>url prefix</dfn>, which is a base url that is appended to the start of all WebDriver URLs. By default this is the empty string.</p>
+
+    <aside class="note">
+      <p>For example a remote end wishing to run alongside other services on <code>example.com</code> might set its <a>url prefix</a> to <code>/wd</code> so that a <a>new session</a> command would be invoked by sending a POST request to <code>/wd/session</code> rather than <code>/session</code>.</p>
+    </aside>
+
+    <p>In order to <dfn>match a request</dfn> given a <var>method</var> and <var>url</var>, the following steps must be taken:</p>
+
+    <ol>
+      <li><p>Let <var>endpoints</var> be a list containing each row in the <a>table of endpoints</a>.</p></li>
```

Comment:
“Lists” have a lot of implementation connotations.  Maybe “sequence” or “array” would be better.